flink-streaming-java/src/main/java/org/apache/flink/streaming/api/environment/CheckpointConfig.java

```diff
@@ -186,7 +164,7 @@ public void disableCheckpointing() {
      * @return True if checkpointing is enables, false otherwise.
      */
     public boolean isCheckpointingEnabled() {
-        return checkpointInterval > 0;
+        return configuration.contains(ExecutionCheckpointingOptions.CHECKPOINTING_INTERVAL);
```

Comment:
Note: long-term we may want to decouple checkpointing being enabled from the interval, in case a user wants to fully control checkpoints from the outside. (Which you currently _can_ by setting to Long.MAX_VALUE but it's a bit weird)

Reply:
How about we introduce a new config option `checkpoint.enabled`? This would fix this issue properly.

Reply:
IIRC we'd have to touch other components as well that expect the interval to be > 0 if checkpointing is enabled. like the checkpoint coordinator maybe?

Reply:
Yes, and I'm -1 for adding an extra `ConfigOption` that would only confuse users and allow for inconsistencies:
- what if user modifies only `checkpoint.enabled` but doesn't set the interval by mistake?
- what if user does the opposite?

If anything, I would be ok with a more fancy single property that allows `manual`, `disabled` or `int` for the interval, but that would be quite a bit of work that's IMO not worth it. `Long.MAX_VALUE` for those two power users that will want to to manually trigger checkpoints is perfectly fine for me.